core/src/main/java/org/apache/carbondata/core/datamap/Segment.java

```diff
@@ -36,9 +39,23 @@
 
   private String segmentFileName;
 
+  private ReadCommittedScope readCommittedScope;
+
   public Segment(String segmentNo, String segmentFileName) {
     this.segmentNo = segmentNo;
     this.segmentFileName = segmentFileName;
+    this.readCommittedScope = null;
+  }
+
+  public Segment(String segmentNo, String segmentFileName, ReadCommittedScope readCommittedScope) {
+    this.segmentNo = segmentNo;
+    this.segmentFileName = segmentFileName;
+    this.readCommittedScope = readCommittedScope;
+  }
+
+
+  public Map<String, String> getCommittedIndexFile() throws IOException {
```

Comment:
add comment for this function, what is the return type (what map to what)

Reply:
Done